src/commonMain/kotlin/fr/acinq/lightning/wire/RouteBlinding.kt

```diff
@@ -22,14 +23,14 @@ sealed class RouteBlindingEncryptedDataTlv : Tlv {
     }
 
     /** Id of the next node. */
-    data class OutgoingNodeId(val nodeId: PublicKey) : RouteBlindingEncryptedDataTlv() {
+    data class OutgoingNodeId(val nodeId: NodeId) : RouteBlindingEncryptedDataTlv() {
```

Comment:
You're modifying the official spec TLV here (tag = 4). We need to get agreement on this change at the spec level for that, otherwise we may have painful compatibility issues to handle in the future...I believe that changing the `outgoing_node_id` TLV to use this new `sciddir_or_pubkey` makes a lot of sense and should be in the BOLTs, don't you? This is an opportunity to create a small spec PR based on `master` that:

- extracts the `sciddir_or_pubkey` from the offers spec PR
- uses it in the `outgoing_node_id` TLV field of onion messages / blinded paths
- adds or modifies a test vector to use this

I expect that this spec PR would be accepted somewhat quickly, and it would make the offers spec PR slightly smaller in scope. It's also important to create that spec PR before people actually start using offers, this way we can just change the TLV without caring about backwards-compat.

Reply:
We can merge that code to both `lightning-kmp` and `eclair` though, even before the spec PR gets accepted, because:

- until we ship bolt 12 support in phoenix, we actually will never create such tlv fields
- on the eclair side, we're just adding support for _reading_ this new case, not writing it, so it won't impact other nodes and we can update the code without caring about backwards-compat

But we should still create a spec PR to get wide support for this feature!